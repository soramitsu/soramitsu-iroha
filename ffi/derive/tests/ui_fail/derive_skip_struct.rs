use getset::{MutGetters, Setters};
use iroha_ffi::ffi_bindgen;
use std::mem::MaybeUninit;

#[ffi_bindgen]
#[derive(Setters, MutGetters)]
#[getset(skip)]
pub struct FfiStruct {
    #[getset(set = "pub", get_mut = "pub")]
    a: u32,
    b: i32,
}

fn main() {
    let s: *mut _ = &mut FfiStruct { a: 42, b: 32 };

    let a = MaybeUninit::<*mut u32>::uninit();
    let b = MaybeUninit::<*const i32>::uninit();

    unsafe {
        FfiStruct__a_mut(s, a.as_mut_ptr());
        let a = &mut *a.assume_init();
        FfiStruct__set_a(s, a);

        FfiStruct__b_mut(s, b.as_mut_ptr());
        let b = &*b.assume_init();
        FfiStruct__set_b(s, b);
    }
}
