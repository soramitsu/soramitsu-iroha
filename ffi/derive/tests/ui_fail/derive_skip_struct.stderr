error[E0425]: cannot find function, tuple struct or tuple variant `FfiStruct__b_mut` in this scope
  --> tests/ui_fail/derive_skip_struct.rs:25:9
   |
5  | #[ffi_bindgen]
   | -------------- similarly named function `FfiStruct__a_mut` defined here
...
25 |         FfiStruct__b_mut(s, b.as_mut_ptr());
   |         ^^^^^^^^^^^^^^^^ help: a function with a similar name exists: `FfiStruct__a_mut`

error[E0425]: cannot find function, tuple struct or tuple variant `FfiStruct__set_b` in this scope
  --> tests/ui_fail/derive_skip_struct.rs:27:9
   |
5  | #[ffi_bindgen]
   | -------------- similarly named function `FfiStruct__set_a` defined here
...
27 |         FfiStruct__set_b(s, b);
   |         ^^^^^^^^^^^^^^^^ help: a function with a similar name exists: `FfiStruct__set_a`
